### PR TITLE
Add support for hierarchical configuration

### LINKED ISSUE #61: Add a global configuration
The global configuration should contain some defaults:
- author/copyright information
- license
- default git host (e.g. https://github.com/sifis-home/{project-name})

And populate the build system metadata (e.g. pyproject.toml or Cargo.toml) and the licensing metadata (.reuse/dep5) accordingly.

[figment](http://crates.io/crates/figment) seems a good crate to implement it in a reduced amount of code.

### PR DESCRIPTION
**Pull request by lu-zero**
TODO:
- [x] Add a built-in configuration
- [x] Read a config.toml
- [x] Merge the two above with the cli with the right priority 
- [x] Set the config path to `$HOME/.config/sifis-generate/config.toml`
- [x] Add a cli option to change the config path
- [x] Document everything 

Fixes #61

### CODE REVIEW

File: src/bin/sifis-generate.rs [hunk 1]
```diff
@@ -84,12 +162,32 @@ enum Cmd {
     Yarn(CommonData),
 }
 
+fn local_config() -> anyhow::Result<PathBuf> {
+    let config_dir = std::env::var("XDG_CONFIG_HOME")
```
Comment (Luni-4) [inline src/bin/sifis-generate.rs:166]: Is this environment variable usable on Windows?

File: src/bin/sifis-generate.rs [hunk 2]
```diff
@@ -103,36 +201,66 @@ fn main() -> anyhow::Result<()> {
...
+            )
+        }
+        ("yarn", matches) => {
+            let config = config
+                .merge(ClapSerialized::<CommonData>::globals(matches.clone()))
+                .select("yarn");
+            let data: CommonData = config.extract()?;
+            Yarn::new().create_ci(&data.name, &data.project_path, &data.license, &data.branch)
+        }
+        _ => unreachable!("unexpected command"),
```
Comment (Luni-4) [inline src/bin/sifis-generate.rs:264]: Is it possible to use an enum in some way to remove this match arm?

**Review by Luni-4 — CHANGES_REQUESTED** [review]
Seems good to me, thank you! 

Just two questions

File: src/bin/sifis-generate.rs (hunk 1 above)
Comment (lu-zero) [reply src/bin/sifis-generate.rs:166]: as long you set it.

**Pull request merged**